Crash RTMP: Add path via API in runOnReady that reads that path

### Which version are you using?

v1.18.1

### Which operating system are you using?

Other (please describe)

### Describe the issue

MediaMTX crashes if you try to add a path from runOnReady hook.
In this case if you add a path that reads the original path. 
It doesn't crash every single time but it should be quite easily reproduced.
This happens on both Linux amd64 as well as Windows

### Describe how to replicate the issue

1. start the MediaMTX
2. publish with rtmp to ``rtmp://127.0.0.1:1935/1``


### MediaMTX configuration

```yaml
# Global settings
logLevel: debug
logDestinations: [stdout]
logStructured: true
logFile: mediamtx.log
dumpPackets: false

readTimeout: 10s
writeTimeout: 10s
writeQueueSize: 2048
udpMaxPayloadSize: 1452
udpReadBufferSize: 0

api: true
apiAddress: :9997
apiEncryption: false
apiAllowOrigins: ["*"]
apiTrustedProxies: []

metrics: false
metricsAddress: :9998
metricsEncryption: false
metricsAllowOrigins: ["*"]
metricsTrustedProxies: []

pprof: false

playback: true
playbackAddress: :9996
playbackEncryption: false
playbackAllowOrigins: ["*"]
playbackTrustedProxies: []

rtsp: false

rtmp: true
rtmpEncryption: "no"
rtmpAddress: :1935

hls: true
hlsAddress: :8888
hlsEncryption: false
hlsAllowOrigins: ["*"]
hlsTrustedProxies: []
hlsAlwaysRemux: true
hlsVariant: lowLatency
hlsSegmentCount: 14
hlsSegmentDuration: 1s
hlsPartDuration: 200ms
hlsSegmentMaxSize: 50M
hlsDirectory: ""
hlsMuxerCloseAfter: 60s

webrtc: false

srt: false

paths:
  1:
    source: publisher
    overridePublisher: false
    record: false
    runOnReady: curl -X POST http://localhost:9997/v3/config/paths/add/2 -d '{"name":"2","source":"rtmp://localhost:1935/1"}'
    runOnReadyRestart: false
  all_others:
```

### MediaMTX logs

{"timestamp":"2026-05-01T07:56:09.4280584+02:00","level":"INF","message":"[RTMP] [conn [::1]:56326] opened"}
{"timestamp":"2026-05-01T07:56:09.4291165+02:00","level":"INF","message":"[RTMP] [conn [::1]:56326] is reading from path '1', 3 tracks (H264, MPEG-4 Audio, MPEG-4 Audio)"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x7ff6a3ed9a7a]

goroutine 86 [running]:
github.com/bluenviron/mediamtx/internal/protocols/rtmp.ToStream(0x74964039f58, 0x749647020a0)
        /s/internal/protocols/rtmp/to_stream.go:42 +0x7a
github.com/bluenviron/mediamtx/internal/staticsources/rtmp.(*Source).runReader(0x74964eea1b0, 0x7496423ae70)
        /s/internal/staticsources/rtmp/source.go:124 +0x136
github.com/bluenviron/mediamtx/internal/staticsources/rtmp.(*Source).Run.func1()
        /s/internal/staticsources/rtmp/source.go:91 +0x25
created by github.com/bluenviron/mediamtx/internal/staticsources/rtmp.(*Source).Run in goroutine 113
        /s/internal/staticsources/rtmp/source.go:90 +0x5df

### Packet dump

_No response_

--- a/reader.go
+++ b/reader.go
@@ -487,6 +487,18 @@ func (r *Reader) readTracks() (map[uint8]*Track, map[uint8]*Track, error) {
 		return nil, nil, fmt.Errorf("no tracks found")
 	}
 
+	for id, track := range videoTracks {
+		if track == nil {
+			return nil, nil, fmt.Errorf("video track %d not configured", id)
+		}
+	}
+
+	for id, track := range audioTracks {
+		if track == nil {
+			return nil, nil, fmt.Errorf("audio track %d not configured", id)
+		}
+	}
+
 	return videoTracks, audioTracks, nil
 }
 

--- a/reader_test.go
+++ b/reader_test.go
@@ -1609,6 +1609,93 @@ func TestReadTracks(t *testing.T) {
 	}
 }
 
+func TestReadTracksErrors(t *testing.T) {
+	for _, ca := range []struct {
+		name     string
+		errMsg   string
+		messages []message.Message
+	}{
+		{
+			"video multitrack coded frames without sequence start",
+			"video track 1 not configured",
+			[]message.Message{
+				&message.VideoExMultitrack{
+					MultitrackType: 0,
+					TrackID:        1,
+					Wrapped: &message.VideoExCodedFrames{
+						ChunkStreamID:   message.VideoChunkStreamID,
+						MessageStreamID: 0x1000000,
+						FourCC:          message.FourCCAVC,
+						DTS:             0,
+					},
+				},
+				&message.VideoExMultitrack{
+					MultitrackType: 0,
+					TrackID:        1,
+					Wrapped: &message.VideoExCodedFrames{
+						ChunkStreamID:   message.VideoChunkStreamID,
+						MessageStreamID: 0x1000000,
+						FourCC:          message.FourCCAVC,
+						DTS:             2 * time.Second,
+					},
+				},
+			},
+		},
+		{
+			"audio multitrack sequence start without coded frames",
+			"audio track 1 not configured",
+			[]message.Message{
+				&message.Video{
+					ChunkStreamID:   message.VideoChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Codec:           message.CodecH264,
+					Type:            message.VideoTypeAU,
+					DTS:             0,
+				},
+				&message.AudioExMultitrack{
+					MultitrackType: 0,
+					TrackID:        1,
+					Wrapped: &message.AudioExSequenceStart{
+						ChunkStreamID:   message.AudioChunkStreamID,
+						MessageStreamID: 0x1000000,
+						FourCC:          message.FourCCMP4A,
+						AACConfig: &mpeg4audio.AudioSpecificConfig{
+							Type:         mpeg4audio.ObjectTypeAACLC,
+							SampleRate:   48000,
+							ChannelCount: 2,
+						},
+					},
+				},
+				&message.Video{
+					ChunkStreamID:   message.VideoChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Codec:           message.CodecH264,
+					Type:            message.VideoTypeAU,
+					DTS:             2 * time.Second,
+				},
+			},
+		},
+	} {
+		t.Run(ca.name, func(t *testing.T) {
+			var buf bytes.Buffer
+			bc := bytecounter.NewReadWriter(&buf)
+			mrw := message.NewReadWriter(bc, bc, true)
+
+			for _, msg := range ca.messages {
+				err := mrw.Write(msg)
+				require.NoError(t, err)
+			}
+
+			c := &dummyConn{rw: &buf}
+			c.initialize()
+
+			r := &Reader{Conn: c}
+			err := r.Initialize()
+			require.EqualError(t, err, ca.errMsg)
+		})
+	}
+}
+
 func TestReaderRewind(t *testing.T) {
 	messages := []message.Message{
 		&message.Video{

